src/vhdlConvertor/typeDeclarationParser.cpp

```diff
@@ -0,0 +1,238 @@
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitScalar_type_definition(
+		std::string &&name, vhdlParser::Scalar_type_definitionContext *ctx) {
+	//scalar_type_definition:
+	//      enumeration_type_definition
+	//      | integer_type_definition
+	//      | floating_type_definition
+	//      | physical_type_definition
+	//;
+	auto e = ctx->enumeration_type_definition();
+	if (e)
+		return visitEnumeration_type_definition(name, e);
+	auto i = ctx->integer_type_definition();
+	if (i)
+		return visitInteger_type_definition(name, i);
+	auto f = ctx->floating_type_definition();
+	if (f)
+		return visitFloating_type_definition(name, f);
+	auto p = ctx->physical_type_definition();
+	if (p)
+		return visitPhysical_type_definition();
+	
+	NotImplementedLogger::print("TypeDeclarationParser.type_definition", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitComposite_type_definition(
+		std::string &&name, vhdlParser::Composite_type_definitionContext *ctx) {
+	//composite_type_definition:
+	//      array_type_definition
+	//      | record_type_definition
+	//;
+	auto arr = ctx->array_type_definition();
+	if (arr)
+		return visitArray_type_definition(name, arr);
+	auto rec = ctx->record_type_definition();
+	if (rec)
+		return visitRecord_type_definition(name, arr);
+
+	NotImplementedLogger::print("TypeDeclarationParser.composite_type_definition", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitAccess_type_definition(
+		std::string &&name, vhdlParser::Access_type_definitionContext *ctx) {
+	
+	NotImplementedLogger::print("TypeDeclarationParser.access_type_definition", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitFile_type_definition(
+		std::string &&name, vhdlParser::File_type_definitionContext *ctx) {
+	
+	NotImplementedLogger::print("TypeDeclarationParser.file_type_definition", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitProtected_type_definition(
+		std::string &&name, vhdlParser::Protected_type_definitionContext *ctx) {
+	
+	NotImplementedLogger::print("TypeDeclarationParser.protected_type_definition", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitEnumeration_type_definition(
+		std::string &&name, vhdlParser::Enumeration_type_definitionContext *ctx) {
+	//enumeration_type_definition:
+	//      LPAREN enumeration_literal ( COMMA enumeration_literal )* RPAREN
+	//;
+	std::unique_ptr<HdlEnumTypeDec> typ = create_object<HdlEnumTypeDec>(ctx, name);
```

Comment:
The rvalue `name` eventually gets used in creating a new AST object like this.  Is this correct and efficient? :-)